### PR TITLE
0.23 -> main

### PR DESCRIPTION
**Pull request by armanbilge**
No conflicts.

### CODE REVIEW

File: server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala [hunk 1]
```diff
@@ -129,9 +129,14 @@ object Throttle {
       Throttle.httpRoutes(bucket, defaultResponse[F] _)(httpRoutes)
     )
 
+  @deprecated("Use httpApp instead.", "0.23.14")
+  def httpAapp[F[_]](amount: Int, per: FiniteDuration)(httpApp: HttpApp[F])(implicit
```
Comment (danicheg) [inline server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala:133]: I think we can remove this

**Review by danicheg — APPROVED** [review]

File: server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala (hunk 1 above)
Comment (armanbilge) [reply server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala:133]: Maybe we should give once cycle of deprecation, to help migration?

Comment (danicheg) [reply server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala:133]: 🤷🏻 Can't suggest how many users might be affected by just removing this, but this was introduced weeks ago

Comment (danicheg) [reply server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala:133]: I don't mind if we release this with deprecation and then remove it. Just we definitely have to remove it for 1.0 :)

**Review by ChristopherDavenport — APPROVED** [review]

**Pull request merged**